libvuln/updates/manager.go

```diff
@@ -0,0 +1,256 @@
+package updates
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strings"
+	"time"
+
+	"golang.org/x/sync/semaphore"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/quay/claircore/internal/vulnstore"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/pkg/distlock/postgres"
+	"github.com/quay/claircore/updater"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	DefaultInterval = time.Duration(30 * time.Minute)
+)
+
+var (
+	DefaultBatchSize = runtime.NumCPU()
+)
+
+type Configs map[string]driver.ConfigUnmarshaler
+
+// Manager oversees the configuration and invocation of vulstore updaters.
+//
+// The Manager may be used in a one-shot fashion, configured
+// to run background jobs, or both.
+type Manager struct {
+	// provides run-time updater construction.
+	factories map[string]driver.UpdaterSetFactory
+	// max in-flight updaters.
+	batchSize int
+	// update interval used once Manager.Start is invoked, otherwise
+	// this field is not used.
+	interval time.Duration
+	// configs provided to updaters once constructed.
+	configs Configs
+	// provided to construct distributed locks.
+	// TODO(louis): this will be replaced by newest distlock implementation.
+	pool   *pgxpool.Pool
```

Comment:
Right now, our distlock implementation is not shareable. A pool needs to be provided so that a distlock can be constructed for every run. Runs may happen in parallel. 

When the new distlock implementation is worked in, this will be replaced with a sharable distlock.